### PR TITLE
Add uploading coverage steps

### PR DESCRIPTION
**Pull request by MaryGao**
ref: https://github.com/Azure/cadl-ranch/blob/main/docs/using-cadl-ranch.md#upload-coverage

the internal ci triggered here: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1908008&view=results

relevant dashboard is shown here: https://azure.github.io/cadl-ranch/

### CODE REVIEW

**Review by qiaozha — APPROVED** [review]

File: packages/cadl-rlc-emitter/coverage/cadl-ranch-coverage-typescript.json [hunk 1]
```diff
@@ -0,0 +1,123 @@
+{
```
[the remark this answers is not in the document]
Comment (MaryGao) [reply packages/cadl-rlc-emitter/coverage/cadl-ranch-coverage-typescript.json:1]: I will remove this once the stop feature is ready

File: .scripts/ci.yml [hunk 2]
```diff
@@ -263,17 +264,26 @@ stages:
...
             continueOnError: true
-          - publish: $(Build.SourcesDirectory)/packages/cadl-rlc-emitter/node_modules/@azure-tools/cadl-ranch-specs/cadl-ranch-coverage-typescript.json
+          - publish: $(CADL_TESTSERVER_COVERAGE_REPORT)
             artifact: CoverageReport
             displayName: "Publish coverage report"
             condition: and(ne(variables['Build.Reason'], 'PullRequest'), eq(variables['System.TeamProject'], 'internal'))
-          - script: npm run check:tree
-            workingDirectory: $(Build.SourcesDirectory)/packages/cadl-rlc-emitter
-            displayName: "Check git Tree"
+          - ${{if and(ne(variables['Build.Reason'], 'PullRequest'), eq(variables['System.TeamProject'], 'internal'))}}:
```
Comment (MaryGao) [inline .scripts/ci.yml:278]: We need to use this way to avoid the permission check in pr ci

File: packages/cadl-rlc-emitter/package.json [hunk 3]
```diff
@@ -11,10 +11,10 @@
     "lint": "eslint src",
     "lint:fix": "eslint src --fix",
     "check:tree": "ts-node ./test/commands/check-clean-tree.ts",
-    "integration-test": "npm run start-test-server & npm run generate-cadl-only && npm run integration-test:alone",
+    "integration-test": "npm run start-test-server & npm run generate-cadl-only && npm run integration-test:alone && npm run stop-test-server",
     "generate-cadl-only": "ts-node ./test/commands/gen-cadl-ranch.ts",
-    "start-test-server": "cd ./node_modules/@azure-tools/cadl-ranch-specs && cadl-ranch serve ./http  --coverageFile ./cadl-ranch-coverage-typescript.json",
-    "coverage":"cd ./node_modules/@azure-tools/cadl-ranch-specs && cadl-ranch check-coverage ./http --coverageFiles ./cadl-ranch-coverage-typescript.json",
+    "start-test-server": "mkdir coverage && npx cadl-ranch serve ./node_modules/@azure-tools/cadl-ranch-specs/http  --coverageFile ./coverage/cadl-ranch-coverage-typescript.json",
+    "stop-test-server": "npx cadl-ranch server stop",
```
Comment (MaryGao) [inline packages/cadl-rlc-emitter/package.json:17]: Use the stop step to save coverage

**Review by qiaozha — APPROVED** [review]

**Pull request merged**